Sources/Sentry/SentrySessionReplay.m

```diff
@@ -306,11 +306,12 @@ - (void)takeScreenshot
         _processingScreenshot = YES;
     }
 
-    UIImage *screenshot = [_screenshotProvider imageWithView:_rootView options:_replayOptions];
-
-    _processingScreenshot = NO;
-
-    [self->_replayMaker addFrameAsyncWithImage:screenshot];
+    [_screenshotProvider imageWithView:_rootView
+                               options:_replayOptions
+                            onComplete:^(UIImage *screenshot) {
+                                self->_processingScreenshot = NO;
+                                [self->_replayMaker addFrameAsyncWithImage:screenshot];
```

Comment:
`m`: Can this lead to a retain cycle? Should we maybe use `__weak` to self here?

Reply:
It can hold the reference for a little longer. But will not create a retain cycle, once the callback returns everything will be disposed.

Reply:
But we call onComplete with a strong reference to self async here https://github.com/getsentry/sentry-cocoa/blob/cb3bca31cafa521d1bfe3e3e627d9284d95c7e6e/Sources/Swift/Tools/SentryViewPhotographer.swift#L35-L45

We have to be careful not to introduce retains cycles in the future. Currently, it doesn't matter much, because we always have a strong reference to `SentrySessionReplay` once session replay is enable. Maybe it's safer to use a `__weak` reference now to avoid problems in the future. Up to you.